ksql-cli/src/main/java/io/confluent/ksql/Ksql.java

```diff
@@ -16,42 +16,64 @@
 
 package io.confluent.ksql;
 
-import com.github.rvesse.airline.help.Help;
-import com.github.rvesse.airline.parser.errors.ParseException;
-import io.confluent.ksql.cli.commands.Local;
-import io.confluent.ksql.cli.commands.Remote;
-import io.confluent.ksql.cli.commands.Standalone;
-import io.confluent.ksql.util.CommonUtils;
+import org.apache.kafka.streams.StreamsConfig;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.Properties;
+
+import io.confluent.ksql.cli.Cli;
+import io.confluent.ksql.cli.Options;
+import io.confluent.ksql.cli.console.JLineTerminal;
+import io.confluent.ksql.rest.client.KsqlRestClient;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.KsqlException;
+import io.confluent.ksql.version.metrics.KsqlVersionCheckerAgent;
+import io.confluent.ksql.version.metrics.collector.KsqlModuleType;
 
 public class Ksql {
 
-  public static void main(String[] args) {
-    Runnable runnable = null;
-    com.github.rvesse.airline.Cli<Runnable> cli =
-        com.github.rvesse.airline.Cli.<Runnable>builder("Cli")
-            .withDescription("Kafka Query Language")
-            .withDefaultCommand(Help.class)
-            .withCommand(Local.class)
-            .withCommand(Remote.class)
-            .withCommand(Standalone.class)
-            .build();
-
-    try {
-      runnable = cli.parse(args);
-      runnable.run();
-    } catch (ParseException exception) {
-      if (exception.getMessage() != null) {
-        System.err.println(exception.getMessage());
-      } else {
-        System.err.println("Options parsing failed for an unknown reason");
-      }
-      System.err.println("See the help command for usage information");
-    } catch (Exception e) {
-      System.err.println(CommonUtils.getErrorMessageWithCause(e));
+  public static void main(String[] args) throws IOException {
+    final Options options = Options.parse(args);
+    if (options == null) {
```

Comment:
Should we print an error message before exit?

Reply:
It gets printed during parsing